presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryResultPageSource.java

```diff
@@ -0,0 +1,342 @@
+        Storage.ReadRowsResponse response = responses.next();
+        Iterable<GenericRecord> records = parse(response);
+        for (GenericRecord record : records) {
+            pageBuilder.declarePosition();
+            for (int column = 0; column < columnTypes.size(); column++) {
+                BlockBuilder output = pageBuilder.getBlockBuilder(column);
+                appendTo(columnTypes.get(column), record.get(column), output);
+            }
+        }
+
+        Page page = pageBuilder.build();
+        pageBuilder.reset();
+        return page;
+    }
+
+    private void appendTo(Type type, Object value, BlockBuilder output)
+    {
+        if (value == null) {
+            output.appendNull();
+            return;
+        }
+
+        Class<?> javaType = type.getJavaType();
+        try {
+            if (javaType == boolean.class) {
+                type.writeBoolean(output, (Boolean) value);
+            }
+            else if (javaType == long.class) {
+                if (type.equals(BIGINT)) {
+                    type.writeLong(output, ((Number) value).longValue());
+                }
+                else if (type.equals(INTEGER)) {
+                    type.writeLong(output, ((Number) value).intValue());
+                }
+                else if (type.equals(DATE)) {
+                    type.writeLong(output, ((Number) value).intValue());
+                }
+                else if (type.equals(TIMESTAMP)) {
+                    type.writeLong(output, BigQueryType.toPrestoTimestamp(((org.apache.avro.util.Utf8) value).toString()));
+                }
+                else if (type.equals(TIME_WITH_TIME_ZONE)) {
+                    type.writeLong(output, DateTimeEncoding.packDateTimeWithZone(((Long) value).longValue() / 1000, TimeZoneKey.UTC_KEY));
+                }
+                else if (type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
+                    type.writeLong(output, DateTimeEncoding.packDateTimeWithZone(((Long) value).longValue() / 1000, TimeZoneKey.UTC_KEY));
+                }
+                else {
+                    throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Unhandled type for %s: %s", javaType.getSimpleName(), type));
+                }
+            }
+            else if (javaType == double.class) {
+                type.writeDouble(output, ((Number) value).doubleValue());
+            }
+            else if (javaType == Slice.class) {
+                writeSlice(output, type, value);
+            }
+            else if (javaType == Block.class) {
+                writeBlock(output, type, value);
+            }
+            else {
+                throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Unhandled type for %s: %s", javaType.getSimpleName(), type));
+            }
+        }
+        catch (ClassCastException ignore) {
+            // returns null instead of raising exception
```

Comment:
throw exception? or warn message?

Reply:
castException can be tolerated,so just return null is fine. warning message usually doesn't get issued from connector right? it's handled either in analyzer/optimizer level. I can add a log message if needed.

Reply:
silently appendNull is not good, when there is cast exception. Let's either throw exception, or at least add warn message

Reply:
on 2nd thought this castException needs to be handled using BigQueryException, fixed